kin-sdk/src/main/java/kin/sdk/TransactionSender.java

```diff
@@ -212,4 +229,14 @@ private boolean isInsufficientKinException(TransactionFailedException transactio
         List<String> resultCodes = transactionException.getOperationsResultCodes();
         return resultCodes != null && resultCodes.size() > 0 && INSUFFICIENT_KIN_RESULT_CODE.equals(resultCodes.get(0));
     }
+
+    private boolean isInsufficientFeeException(TransactionFailedException transactionException) {
+        String transactionResultCode = transactionException.getTransactionResultCode();
+        return !TextUtils.isEmpty(transactionResultCode) && INSUFFICIENT_FEE_RESULT_CODE.equals(transactionResultCode);
+    }
+
+    private boolean isInsufficientBalanceException(TransactionFailedException transactionException) {
```

Comment:
why we need a new exception? seems the same logically to `InsufficientKinException`, probably just the underlined error code are different between asset and native?